lib/routes/route.js

```diff
@@ -612,6 +612,35 @@ module.exports = function(app, router) {
         });
     });
 
+    router.get("/user", function (req, res, next) {
+        handleSession(req.session, function (err, sessionObj) {
+            if (err) {
+                if (err.type === "userID") {
+                    req.session.userID = undefined;
+                    sessionObj = {};
```

Comment:
```suggestion
                var err = {
                    message: "Cannot perform action. User no longer exists."
                };
                return routeError(err, next, {
                    errorID: "sessionUserNotFound",
                    statusCode: 404
                });
```
If I'm not mistaken, if `err.type` is `"userID"` then that means the current session user no longer exists in the database. I think we should instead return error in this case.